rust/rsmgp-sys/src/value/mod.rs

```diff
@@ -0,0 +1,370 @@
+                ffi::mgp_value_destroy(self.ptr);
+            }
+        }
+    }
+}
+
+impl MgpValue {
+    pub fn new(ptr: *mut mgp_value) -> MgpValue {
+        MgpValue { ptr }
+    }
+
+    pub(crate) fn mgp_ptr(&self) -> *const mgp_value {
+        self.ptr
+    }
+
+    pub fn to_value(&self, memgraph: &Memgraph) -> MgpResult<Value> {
+        unsafe { mgp_raw_value_to_value(self.mgp_ptr(), &memgraph) }
+    }
+
+    pub fn make_null(memgraph: &Memgraph) -> MgpResult<MgpValue> {
+        unsafe {
+            let mgp_ptr = ffi::mgp_value_make_null(memgraph.memory());
+            if mgp_ptr.is_null() {
+                return Err(MgpError::UnableToMakeNullValue);
+            }
+            Ok(MgpValue::new(mgp_ptr))
+        }
+    }
+
+    pub fn is_null(&self) -> bool {
+        unsafe { ffi::mgp_value_is_null(self.ptr) != 0 }
+    }
+
+    pub fn make_bool(value: bool, memgraph: &Memgraph) -> MgpResult<MgpValue> {
+        unsafe {
+            let mgp_ptr = ffi::mgp_value_make_bool(if !value { 0 } else { 1 }, memgraph.memory());
+            if mgp_ptr.is_null() {
+                return Err(MgpError::UnableToMakeBoolValue);
+            }
+            Ok(MgpValue::new(mgp_ptr))
+        }
+    }
+
+    pub fn is_bool(&self) -> bool {
+        unsafe { ffi::mgp_value_is_bool(self.ptr) != 0 }
+    }
+
+    pub fn make_int(value: i64, memgraph: &Memgraph) -> MgpResult<MgpValue> {
+        unsafe {
+            let mgp_ptr = ffi::mgp_value_make_int(value, memgraph.memory());
+            if mgp_ptr.is_null() {
+                return Err(MgpError::UnableToMakeIntegerValue);
+            }
+            Ok(MgpValue::new(mgp_ptr))
+        }
+    }
+
+    pub fn is_int(&self) -> bool {
+        unsafe { ffi::mgp_value_is_int(self.ptr) != 0 }
+    }
+
+    pub fn make_double(value: f64, memgraph: &Memgraph) -> MgpResult<MgpValue> {
+        unsafe {
+            let mgp_ptr = ffi::mgp_value_make_double(value, memgraph.memory());
+            if mgp_ptr.is_null() {
+                return Err(MgpError::UnableToMakeDoubleValue);
+            }
+            Ok(MgpValue::new(mgp_ptr))
+        }
+    }
+
+    pub fn is_double(&self) -> bool {
+        unsafe { ffi::mgp_value_is_double(self.ptr) != 0 }
+    }
+
+    pub fn make_string(value: &CStr, memgraph: &Memgraph) -> MgpResult<MgpValue> {
+        unsafe {
+            let mgp_ptr = ffi::mgp_value_make_string(value.as_ptr(), memgraph.memory());
+            if mgp_ptr.is_null() {
+                return Err(MgpError::UnableToMakeMemgraphStringValue);
+            }
+            Ok(MgpValue::new(mgp_ptr))
+        }
+    }
+
+    pub fn is_string(&self) -> bool {
+        unsafe { ffi::mgp_value_is_string(self.ptr) != 0 }
+    }
+
+    /// Required because List implements Drop which destroys the underlying list object. MgpValue
+    /// takes the ownership of the list which means copy has to be created + if all goes well the
+    /// list object shouldn't be deleted.
```

Comment:
What do you mean by "if all goes well, the list object shouldn't be deleted"? Is this about the `list` argument? because then why would that get deleted? It this is about the new object, then this comment is kind of unnecessary in my opinion.

Reply:
Yes, I've added this comment because we can't just wrap the mgp_list pointer to the `List` because then it will be automatically deleted, but yea, the `if all goes well` part is strange. I'll try to rephrase.